cmd/cloud-api-adaptor/main.go

```diff
@@ -45,8 +47,8 @@ var hypcfg hypervisor.Config
 
 func (cfg *daemonConfig) Setup() (cmd.Starter, error) {
 
-	if len(os.Args) < 2 {
-		fmt.Printf("%s aws|azure|ibmcloud|libvirt <options>\n", os.Args[0])
+	if len(os.Args) < 2 || !contains(os.Args[1], supportedProviders) {
+		fmt.Printf("Invalid or no cloud provider passed.\n%s aws|azure|ibmcloud|libvirt|vsphere <options>\n", os.Args[0])
```

Comment:
Can we keep a single list of strings to maintain?
Something like: 

```suggestion
		fmt.Printf("Invalid or no cloud provider passed.\n%s %s <options>\n", os.Args[0], strings.Join(supportedProviders, "|"))
```

Reply:
Changed it.